src/amo/components/RatingManager/index.js

```diff
@@ -276,25 +283,53 @@ export const mapStateToProps = (state: AppState, ownProps: Props) => {
 export const mapDispatchToProps = (
   dispatch: DispatchFunc,
 ): DispatchMappedProps => ({
-  loadSavedReview({ apiState, userId, addonId, versionId }) {
+  loadSavedReview({ apiState, userId, addonId, addonSlug, versionId }) {
     return getLatestUserReview({
       apiState,
       user: userId,
       addon: addonId,
       version: versionId,
     }).then((review) => {
       if (review) {
-        dispatch(setReview(review));
+        dispatch(
+          setLatestReview({ userId, addonId, addonSlug, versionId, review }),
+        );
       } else {
-        log.info(
+        log.debug(
           `No saved review found for userId ${userId}, addonId ${addonId}`,
         );
+        dispatch(
+          setLatestReview({
+            userId,
+            addonId,
+            addonSlug,
+            versionId,
+            review: null,
+          }),
+        );
       }
     });
   },
 
   submitReview(params) {
-    return submitReview(params).then((review) => dispatch(setReview(review)));
+    return submitReview(params).then((review) => {
+      // The API could possibly return a null review.version if that
+      // version was deleted. In that case, we fall back to the submitted
+      // versionId which came from the page data. It is highly unlikely
```

Comment:
Confidence? Hopefully 100% because we'll never get the information in -prod since `invariant` calls are stripped.

Reply:
I'm 99% confident we'll never get it (on this page anyway) since there is no way to load a deleted version. The code has to be defensive here only because the Flow type says it's *possible* which is technically true.